src/lib.rs

```diff
@@ -28,10 +28,10 @@ pub static STANDARD: Config = Config {char_set: CharacterSet::Standard, pad: tru
 pub static URL_SAFE: Config = Config {char_set: CharacterSet::UrlSafe, pad: true};
 pub static URL_SAFE_NO_PAD: Config = Config {char_set: CharacterSet::UrlSafe, pad: false};
 
-
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug)]
 pub enum Base64Error {
     Utf8(str::Utf8Error),
+    FromUtf8(string::FromUtf8Error),
```

Comment:
I'm not sure this is a good idea... Now, someone who's not using the traits still has to handle this error case even though it's impossible in their usage, which isn't ideal. TBH I don't find these sorts of convenience traits all that convenient as they tend to create project-specific API of things like String based on what traits are mixed on, and that inhibits new developers coming in to a project and things like that, but I don't mind them as long as they're opt-in. (Also, they encourage the allocation-heavy flavor of base64 usage, which makes me wipe away a tiny tear.) However, the presence of this error variant means that now everyone has to care/know about the extra traits even when they don't actually use them.

In fact, I'm pretty sure we can ditch the existing `Utf8` error variant as well since we create (correct) UTF-8 via the powers of unsafe. ;)

It looks like this is only even needed because of `FromBase64` for `[u8]`. Rather than having the intervening string and UTF-8, we should just expose a way to decode bytes. (It already is just handling bytes under the hood, anyway.) Hmm... perhaps `decode_config_buf` should have its input be bytes? Or have another similar function, though we already have so many `decode_*`. WDYT?